src/vs/workbench/contrib/positronPlots/browser/positronPlotsActions.ts

```diff
@@ -151,85 +273,33 @@ export class PlotsCopyAction extends Action2 {
-			if (editorService.activeEditorPane?.getId() === PositronPlotsEditorInput.EditorID) {
-				const plotId = editorService.activeEditorPane?.input?.resource?.path.toString();
-				if (plotId) {
-					this.copyEditorPlotToClipboard(plotsService, notificationService, editorService, plotId);
-				}
+	override executeQuickPick(selectedItem: IQuickPickItem, plotsService: IPositronPlotsService,
+		editorService: IEditorService, notificationService: INotificationService): Promise<void> {
+		if (selectedItem?.id) {
+			if (selectedItem.id === PlotActionTarget.VIEW) {
+				this.copyViewPlotToClipboard(plotsService, notificationService);
 			} else {
-				notificationService.error(localize('positronPlots.editorCopyNotActive', 'Active editor is not a plot.'));
+				this.copyEditorPlotToClipboard(plotsService, notificationService, editorService, selectedItem.id);
 			}
-		} else {
-			this._currentQuickPick = quickPick.createQuickPick();
-
-			this._currentQuickPick.items = quickPickItems;
-			this._currentQuickPick.ignoreFocusOut = true;
-			this._currentQuickPick.hideInput = true;
-			this._currentQuickPick.title = localize('positronPlots.copyQuickPickTitle', 'Select the plot to copy to clipboard');
-
-			this._currentQuickPick.onDidAccept((_event) => {
-				const selectedItem = this._currentQuickPick?.selectedItems[0];
-				if (selectedItem) {
-					this.executeCopyPlot(selectedItem, plotsService, notificationService, editorService);
-				} else {
-					notificationService.info(localize('positronPlots.noPlotSelected', 'No plot selected.'));
-				}
-				this._currentQuickPick?.hide();
-			});
-
-			this._currentQuickPick.onDidHide(() => {
-				this._currentQuickPick?.dispose();
-			});
-
-			this._currentQuickPick.onDispose(() => {
-				this._currentQuickPick = undefined;
-			});
-
-			this._currentQuickPick.show();
 		}
+
+		return Promise.resolve();
 	}
 
-	private executeCopyPlot(selectedItem: IQuickPickItem, plotsService: IPositronPlotsService, notificationService: INotificationService, editorService: IEditorService) {
-		if (selectedItem?.id) {
-			if (selectedItem.id === CopyPlotTarget.VIEW) {
-				this.copyViewPlotToClipboard(plotsService, notificationService);
+	override executeTargetAction(target: PlotActionTarget, plotsService: IPositronPlotsService,
+		editorService: IEditorService, notificationService: INotificationService): Promise<void> {
+		if (target === PlotActionTarget.VIEW) {
+			this.copyViewPlotToClipboard(plotsService, notificationService);
+		} else if (target === PlotActionTarget.ACTIVE_EDITOR) {
+			const plotId = this.getActiveEditorPlotId(editorService);
+			if (plotId) {
+				this.copyEditorPlotToClipboard(plotsService, notificationService, editorService, plotId);
 			} else {
-				this.copyEditorPlotToClipboard(plotsService, notificationService, editorService, selectedItem.id);
+				notificationService.info(localize('positronPlots.noPlotSelected', 'No plot selected.'));
 			}
 		}
+
+		return Promise.resolve();
```

Comment:
From a quick parse it doesn't look like we have explicitly async code in this file -- do we need the promise pattern in this file?